### ISSUE TITLE
libnewt: whiptcl.so not installed on 10.13

### ISSUE THREAD

**Issue by nieder** [fink/fink-distributions#162]
On 10.13, whiptcl.so is not being installed and causes a failure when populating the `newt-tcl` SplitOff
````
mv /sw/src/fink.build/root-libnewt0-shlibs-0.52.19-1.1/sw/lib/whiptcl.so /sw/src/fink.build/root-newt-tcl-0.52.19-1.1/sw/lib/whiptcl/
mv: rename /sw/src/fink.build/root-libnewt0-shlibs-0.52.19-1.1/sw/lib/whiptcl.so to /sw/src/fink.build/root-newt-tcl-0.52.19-1.1/sw/lib/whiptcl/whiptcl.so: No such file or directory
````
whiptcl.so is built, just not installed. This is the relevant part of the install process:
````
ln -sf libnewt.0.52.19.dylib /sw/src/fink.build/root-libnewt0-shlibs-0.52.19-1.1//sw/lib/libnewt.0.dylib
[ -n "" ] && install -m 755 whiptcl.so /sw/src/fink.build/root-libnewt0-shlibs-0.52.19-1.1//sw/lib || :
````
On 10.11, that initial comparison is and it succeeds in putting whiptcl.so into %i/lib
````
[ -n "whiptcl.so" ] && install -m 755 whiptcl.so /sw/build.build/root-libnewt0-shlibs-0.52.19-1.1//sw/lib || :
````

**Comment by dmacks** [comment]
Earlier on,  I see:
```
./autogen.sh
configure.ac: error: no proper invocation of AM_INIT_AUTOMAKE was found.
configure.ac: You should verify that configure.ac invokes AM_INIT_AUTOMAKE,
configure.ac: that aclocal.m4 is present in the top-level directory,
configure.ac: and that aclocal.m4 was recently regenerated (using aclocal)
automake: error: no 'Makefile.am' found for any configure output
```
which means any patches against configure.ac and Makefile.am are likely being ignored.

But also:
```
checking for tcl configuration... NOT found
```
It makes a bit of sense that failing to detect some piece of tcl would inhibit complete installation of some tcl thing. The configure script has all kinds of hardcoded guessing of tcl paths that seem likely to mix up apple's and fink's. One of those looks for tclConfig.sh, which *until 10.13* was in /usr/lib, but now is not (/System/Library/Frameworks/Tcl.framework is a backward-compatible location of it).

**Comment by dmacks** [comment]
Passing ```ac_cv_c_tclconfig=%p/lib``` via ConfigureParams and scrapping ```SetCFLAGS: -I/usr/include/tcl8.6``` gets be a fully built package on 10.13.

**Comment by dmacks** [comment]
...and whiptcl.so is also miscompiled (as a runtime program rather than a shared library!) because PatchScript removes its ```-shared``` flag. And also inserts ```-undefined dynamic_lookup```, which masks the fact that it's incomplete in terms of its ABI when the -shared flag is omitted. And it's placed in a subdir but without a pkgIndex.tcl, which makes it hard for tcl to find it. There are lots of upstream and a few packaging mistakes here.

**Comment by dmacks** [comment]
There's a slightly newer version upstream, haven't had a chance to look at it. I filed several upstream bugs against the problems that made our packaging difficult/fragile (linked in the .info I committed).

**Comment by dhomeier** [comment]
I did build `0.52.20` on 10.13, but my 10.13 system is broken, so I cannot check exactly how it went right now. The update is in #204.

**Comment by dmacks** [comment]
It looks easy to move newt-pyXX into a separate varianted .info (getting it out of the C library package-suite altogether). But the existing newt-pyXX have ```Depends:libnewt0-shlibs (=%v-%r)``` which I think can lead to a deadlock when a user who has a newt-pyXX installed tries to update (now that they become separate .info, the "=" makes it hard to find an order in which to install the new %r without leaving the system in an inconsistent state).

**Comment by TheSin-** [comment]
I do not want to separate out the py pacakge, it'll pull it too far away from Debian and create yet an other package to maintain that isn't really different at all just more source to keep in sync.

**Comment by TheSin-** [comment]
I'm still on 10.11 cause I'm stuck here till I have time to port all my fink changes up and dpkg/apt.  So I can't really help with 10.13 stuff sadly.

**Comment by dhomeier** [comment]
My High Sierra machine is back alive for the moment, and I can confirm that `newt-tcl` correctly detects and installs `whiptcl.so` with Xcode 9.4.1
```
/sw/lib/whiptcl/whiptcl.so: Mach-O 64-bit bundle x86_64
```
both for 0.52.19 and the 0.52.20 update, so that seems solved.

As for separating out the package, it would mean a complete rebuild of the source with the patched `Makefile.in` for each `newt-pyXX`, right? The way it's done now creates a dependency of the base package on all 4 pythons, but maybe that's the lesser evil; and staying close to the Debian setup sounds like a valid argument to me as well.

**Comment by TheSin-** [comment]
is this still an issue or can this be closed?

**Comment by nieder** [comment]
No it's not. Dunno if it's my Xcode version or fixes from #163, but it now fully built on 10.13.
